# House Rocket 1

A House Rocket é uma plataforma digital que tem como modelo de negócio, a compra e a venda de imóveis usando tecnologia.
Você é um Data Scientist contrato pela empresa para ajudar a encontrar as melhores oportunidades de negócio no mercado de imóveis. O CEO da House Rocket gostaria de maximizar a receita da empresa encontrando boas oportunidades de negócio.
Sua principal estratégia é comprar boas casas em ótimas localizações com preços baixos e depois revendê-las posteriormente à preços mais altos. Quanto maior a diferença entre a compra e a venda, maior o lucro da empresa e portanto maior sua receita.
Entretanto, as casas possuem muitos atributos que as tornam mais ou menos atrativas aos compradores e vendedores e a localização e o período do ano também podem influenciar os preços.

Nesse primeiro momento, vamos responder as seguintes dúvidas do CEO:

#### 1.Quantas casas estão disponíveis para compra?

#### 2.Quantos atributos as casas possuem?

#### 3.Quais são os atributos das casas?

#### 4.Qual a casa de maior valor?

#### 5.Qual a casa de menor valor?

## Importando as bibliotecas

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

%matplotlib inline

## Obtendo os dados

In [2]:
df = pd.read_csv(r"C:\Users\herna\OneDrive\Lives Python - Meigarom\Live #2\kc_house_data.csv")

In [3]:
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

# Explorando os dados

In [4]:
df.columns

Index(['id', 'date', 'price', 'bedrooms', 'bathrooms', 'sqft_living',
       'sqft_lot', 'floors', 'waterfront', 'view', 'condition', 'grade',
       'sqft_above', 'sqft_basement', 'yr_built', 'yr_renovated', 'zipcode',
       'lat', 'long', 'sqft_living15', 'sqft_lot15'],
      dtype='object')

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

In [6]:
df[['price']].max()

price    7700000.0
dtype: float64

In [7]:
df[['price']].min()

price    75000.0
dtype: float64

## Conclusão 1

1. 21613 casas estão disponíveis para serem compradas.

2. Desconsiderando as colunas de ID e de Data, as casa possuem 19 atributos.

3. Esses atributos são: 'price', 'bedrooms', 'bathrooms', 'sqft_living','sqft_lot', 'floors', 'waterfront', 'view', 'condition', 'grade','sqft_above', 'sqft_basement', 'yr_built', 'yr_renovated', 'zipcode','lat', 'long', 'sqft_living15', 'sqft_lot15'.
       
4. O maior valor é US7700000.

5. O menor valor é US75000.



## Novas perguntas do CEO

#### 6.Qual a soma total de quartos do conjunto de dados?

In [8]:
df[['bedrooms']].sum()

bedrooms    72854
dtype: int64

#### 7.Quantas casas possuem 2 banheiros?

In [9]:
df[df['bathrooms'] == 2].shape

(1930, 21)

#### 8.Qual o preço médio de todas as casas no conjunto de dados?

In [11]:
df['price'].mean()

540088.1417665294

#### 9.Qual o preço médio de casas com 2 banheiros?

In [13]:
df.loc[df['bathrooms'] == 2, 'price'].mean()

457889.7186528497

#### 10.Qual o preço mínimo entre as casas com 3 quartos?

In [14]:
df.loc[df['bathrooms'] == 3, 'price'].min()

154000.0

#### 11.Quantas casas possuem mais de 300 metros quadrados na sala de estar? OBS: 1 pé quadrado = 0,09 metros quadrados

In [16]:
df['m2_living'] = df['sqft_living'] * 0.092

df[df['m2_living']>300].shape

(2141, 22)

#### 12.Quantas casas tem mais de 2 andares?

In [17]:
df[df['floors']>2].shape

(782, 22)

#### 13.Quantas casas tem vista para o mar?

In [18]:
df[df['waterfront'] == 1].shape

(163, 22)

#### 14.Das casas com vista para o mar, quantas tem 3 quartos?

In [20]:
df[(df['waterfront']==1) & (df['bedrooms']==3)].shape

(64, 22)

#### 15.Das casas com mais de 300 metros quadrados de sala de estar, quantas tem mais de 2 banheiros?

In [22]:
df[(df['m2_living']>300) & (df['bathrooms']>2)].shape

(2088, 22)